deepchem/models/tests/test_torch_model.py

```diff
@@ -517,30 +517,36 @@ def test_torch_compile():
                           size=(n_samples, n_tasks)).astype(np.float32)
     dataset = dc.data.NumpyDataset(X, y)
 
-    # Testing backends
-    inductor_model = dc.models.torch_models.CNN(n_tasks,
-                                                n_features,
-                                                dims=1,
-                                                kernel_size=3,
-                                                mode='classification')
-    inductor_model.compile(backend='inductor')
-    inductor_model.fit(dataset, nb_epoch=500)
-
-    inductor_model_output = inductor_model.predict_on_batch(X)
-    inductor_model_output = np.argmax(inductor_model_output, axis=2)
+    def test_config_overfit(mode='default', backend='inductor'):
```

Comment:
A test within a test is not a good pattern. An unit-test must test for only one condition at a time. Maybe you can use pytest parameterize.

Reply:
If we remove the tests for other modes and test only the API of `torch.compile` instead, we can remove the subfunction and the test will only check for one condition. So I'll change the test according to how I'm resolving the previous comment.